scalyr_agent/copying_manager/worker.py

```diff
@@ -1052,22 +1088,49 @@ def _write_full_checkpoint_state(self, current_time):
         from where we left off copying each file.
 
         """
-        self.__write_checkpoint_state(
-            self.__log_processors,
-            "%s%s.json" % (WORKER_SESSION_CHECKPOINT_FILENAME_PREFIX, self._id),
+
+        log_processors = self.__log_processors + self._closed_log_processors
+
+        checkpoints = self._get_log_processors_checkpoints(log_processors, current_time)
+
+        for processor in self.__log_processors:
+            # set this log processor inactive, so it won't be included into the "active" checkpoints
+            # until it is marked as active again.
+            processor.set_inactive()
```

Comment:
If a processor has been closed, is it automatically considered inactive?  I just ask since this is only iterating over `__log_processors` and not `log_processors`.

Reply:
Yes, there has to be a `log_processors ` variable. Thanks.